backend/cfg/cfg_liveness.ml

```diff
@@ -0,0 +1,99 @@
+[@@@ocaml.warning "+a-4-30-40-41-42"]
+
+module Domain = struct
+  type t = Reg.Set.t
+
+  let bot = Reg.Set.empty
+
+  let compare = Reg.Set.compare
+
+  let join = Reg.Set.union
+
+  let less_equal = Reg.Set.subset
+
+  (* CR xclerc for xclerc: already defined elsewhere? *)
+  let with_formatter ~f x =
+    let buff = Buffer.create 64 in
+    let fmt = Format.formatter_of_buffer buff in
+    f fmt x;
+    Format.pp_print_flush fmt ();
+    Buffer.contents buff
+
+  let to_string regset =
+    regset |> Reg.Set.elements
+    |> ListLabels.map ~f:(with_formatter ~f:Printmach.reg)
+    |> StringLabels.concat ~sep:", "
+end
+
+module Transfer = struct
+  type domain = Domain.t
+
+  let basic : domain -> exn:domain -> Cfg.basic Cfg.instruction -> domain =
+   fun value ~exn instr ->
+    match instr.desc with
+    | Op _ | Call _ ->
```

Comment:
The case for `Op` is missing an optimization  that is present in `liveness.ml`: if operation is pure and its result is unused,  then it's arguments are not propagated. Is it intentional?

Reply:
It is an oversight; but the fact that it did not expose
a difference (CI) is an interesting data point.

Reply:
I think it's because the checking is after register allocation, so the dead instructions would have already been eliminated by deadcode pass earlier.